src/app-layer-protos.c

```diff
@@ -25,122 +25,52 @@
 #include "suricata-common.h"
 #include "app-layer-protos.h"
 
-#define CASE_CODE(E)  case E: return #E
+const char *AppProtoStrings[ALPROTO_MAX] = {
+    "unknown",        // ALPROTO_UNKNOWN,
+    "http",           // ALPROTO_HTTP1
```

Comment:
The case statement had the safety that a refactor/re-ordering would keep the values consistent.  How do we ensure that now?

Reply:
That is a good point.

We can have a visual verification by making this a list of tuple, AppProto + string, and debug validate that the index is the value of the AppProto.

What do you think of it ?